### ISSUE TITLE
unexpected keyword argument 'max_retries' on matrix-puppet-hangouts status message after starting

### ISSUE THREAD

**Issue by MurzNN** [matrix-hacks/matrix-puppet-hangouts#7]
After starting puppet - I got in status messages:
```
Starting hangouts bridge! ​
Attempting main loop...
 ​Traceback (most recent call last):
  File "hangups_client.py", line 252, in 
 ​   run_example(listen_events)
  File "hangups_client.py", line 71, in run_example
    client = hangups.Client(cookies, max_retries=float('inf'), retry_backoff_base=1.2)
TypeError: __init__() got an unexpected keyword argument 'max_retries'
​```
And seems it not working after this - I not receive messages from Hangouts to Matrix.

**Comment by tfreedman** [comment]
I get the exact same thing in the Hangouts Protocol room whenever I attempt to run this:

Starting hangouts bridge! ﻿
Attempting main loop...
 ﻿
Traceback (most recent call last):
  File "hangups_client.py", line 252, in 
 ﻿    run_example(listen_events)
  File "hangups_client.py", line 71, in run_example
    client = hangups.Client(cookies, max_retries=float('inf'), retry_backoff_base=1.2)
TypeError: __init__() got an unexpected keyword argument 'max_retries'
 ﻿

**Comment by MurzNN** [comment]
Also I see another keyword argument in logs:
```
    run_example(listen_events)
  File "hangups_client.py", line 71, in run_example
    client = hangups.Client(cookies, max_retries=float('inf'), retry_backoff_base=1.2)
TypeError: __init__() got an unexpected keyword argument 'retry_backoff_base'
```

**Comment by AndrewJDR** [comment]
Sorry for not getting back to you guys earlier on this. The problem is a simple one: You need a newer version of hangups. I don't think @tdryer has gotten around to making a new release yet, so you'll need to grab the git version of hangups and run the bridge using that. To do so, you need to clone https://github.com/tdryer/hangups and run that in a venv, then launch the bridge as usual.

If you're not up for all that, you can just run an older commit of matrix-puppet-hangouts, such as 75bf56e5776ec5af7719aa9d589502571644cf9c. The downside of this approach is that it's not going to be as reliable when it comes to reconnecting to the hangouts server when you get disconnected.

@tdryer, any chance a new hangups release can happen at some point to make this process easier for folks?

**Comment by MurzNN** [comment]
@AndrewJDR thanks for reply, can you describe more details about how to run `hangups` in `venv`? As I understand, `hangups_client.py` must use `hangups` from `pip` that have older version.

**Comment by MurzNN** [comment]
Also after revert back to commit `75bf56e` - I got the errors too:
```
Traceback (most recent call last):
  File "hangups_client.py", line 252, in 
    run_example(listen_events)
  File "hangups_client.py", line 71, in run_example
    client = hangups.Client(cookies, max_retries=float('inf'), retry_backoff_base=1.2)
TypeError: __init__() got an unexpected keyword argument 'retry_backoff_base'
```

**Comment by AndrewJDR** [comment]
@MurzNN 
> Also after revert back to commit 75bf56e - I got the errors too:

If you look at hangups_client.py at that 75bf56e, you'll see it doesn't contain any reference to max_retries, so you must've executed your git commands incorrectly:
https://github.com/matrix-hacks/matrix-puppet-hangouts/blob/75bf56e5776ec5af7719aa9d589502571644cf9c/hangups_client.py
You'll want to double check that you ran your git commands correctly (e.g. you could use something like `git reset --hard 75bf56e`). 

My script to build a venv looks like this (should be run from inside your hangups git checkout):
```
#!/bin/bash                                                                                                                                           
python3 -m venv venv
venv/bin/pip install --upgrade pip 
venv/bin/python3 setup.py install
```

then I activate the venv like this from bash:
```
source <your hangups dir>/venv/bin/activate
```

Then immediately after that, I run matrix-puppet-hangouts as usual.

**Comment by tdryer** [comment]
> @tdryer, any chance a new hangups release can happen at some point to make this process easier for folks?

Sure. Released hangups 0.4.3.

**Comment by MurzNN** [comment]
@tdryer Thanks, updated via pip and all works well!
